lifelines/estimation.py

```diff
@@ -1152,6 +1158,10 @@ def _newton_rhaphson(self, X, T, E, initial_beta=None, step_size=1.,
                                    include_likelihood=include_likelihood)
             # Do not override hessian and gradient in case of garbage
             h, g = output[:2]
+            if self.penalizer > 0:
+                # add the gradient and hessian of the l2 term
+                g -= self.penalizer * beta.T
```

Comment:
You are right, that is the objective function. However, when you differentiate w.r.t. `beta`, the square term is dropped. That's why it's not present here.